internal/engine/experiment/hhfm/hhfm.go

```diff
@@ -322,7 +321,7 @@ func (d Dialer) DialContext(ctx context.Context, network, address string) (net.C
 	if child == nil {
 		// TODO(bassosimone): figure out why using dialer.New here
 		// causes the experiment to fail with eof_error
-		child = &net.Dialer{Timeout: 15 * time.Second}
+		child = netxlite.DefaultDialer
```

Comment:
This is probably good but (1) we should probably construct a netxlite dialer and (2) address the comment or, alternatively, use a `SimpleDialer` and continue using `net.Dialer`.